android/src/main/java/com/pagopa/ioreactnativeintegrity/IoReactNativeIntegrityModule.kt

```diff
@@ -1,25 +1,315 @@
 package com.pagopa.ioreactnativeintegrity
 
+import android.content.pm.PackageManager
+import android.os.Build
+import android.security.keystore.KeyGenParameterSpec
+import android.security.keystore.KeyInfo
+import android.security.keystore.KeyProperties
+import android.security.keystore.KeyProperties.SECURITY_LEVEL_STRONGBOX
+import android.security.keystore.KeyProperties.SECURITY_LEVEL_TRUSTED_ENVIRONMENT
+import android.security.keystore.KeyProperties.SECURITY_LEVEL_UNKNOWN_SECURE
+import android.util.Base64
+import androidx.annotation.RequiresApi
+import com.facebook.react.bridge.Promise
 import com.facebook.react.bridge.ReactApplicationContext
 import com.facebook.react.bridge.ReactContextBaseJavaModule
 import com.facebook.react.bridge.ReactMethod
-import com.facebook.react.bridge.Promise
+import com.facebook.react.bridge.WritableMap
+import com.facebook.react.bridge.WritableNativeMap
+import com.google.android.gms.common.GoogleApiAvailability
+import com.google.android.gms.tasks.Task
+import com.google.android.play.core.integrity.IntegrityManagerFactory
+import com.google.android.play.core.integrity.StandardIntegrityManager
+import com.google.android.play.core.integrity.StandardIntegrityManager.StandardIntegrityToken
+import com.google.android.play.core.integrity.StandardIntegrityManager.StandardIntegrityTokenProvider
+import com.google.android.play.core.integrity.StandardIntegrityManager.StandardIntegrityTokenRequest
+import java.security.KeyFactory
+import java.security.KeyPair
+import java.security.KeyPairGenerator
+import java.security.KeyStore
+import java.security.PrivateKey
+import java.security.spec.ECGenParameterSpec
 
+/**
+ * React Native bridge for Google Play Integrity API and key attestation.
+ * @property integrityTokenProvider the token integrity provider which should be initialized by calling [prepareIntegrityToken]
+ * @property keyStore the key store instance used to generate hardware backed keys and get a key attestation via [getAttestation]
+ */
 class IoReactNativeIntegrityModule(reactContext: ReactApplicationContext) :
   ReactContextBaseJavaModule(reactContext) {
 
+  private lateinit var integrityTokenProvider: StandardIntegrityTokenProvider
```

Comment:
Using `lateinit` force you to catch `UninitializedPropertyAccessException`. If we use this property in a single place could be a good choice, but if we then use it in multiple place could lead in a missing catch. Another option is leverage Kotlin null safety by doing:

```suggestion
  private var integrityTokenProvider: StandardIntegrityTokenProvider? = null
```

And then use the '?' when we use this property. What do you think?

Reply:
In this latter scenario we could change `requestIntegrityToken` this way (for example):

```
fun requestIntegrityToken(requestHash: String?, promise: Promise) {
    try {
      val integrityTokenResponse = integrityTokenProvider?.request(
        StandardIntegrityTokenRequest.builder()
          .setRequestHash(requestHash)
          .build()
      )
      integrityTokenResponse?.apply {
        addOnSuccessListener { res -> promise.resolve((res.token())) }
        addOnFailureListener { ex ->
          ModuleException.REQUEST_TOKEN_FAILED.reject(
            promise,
            Pair(ERROR_USER_INFO_KEY, getExceptionMessageOrEmpty(ex))
          )
        }
      } ?: ModuleException.PREPARE_NOT_CALLED.reject(promise)
    } catch (e: Exception) {
      ModuleException.REQUEST_TOKEN_FAILED.reject(
        promise,
        Pair(ERROR_USER_INFO_KEY, getExceptionMessageOrEmpty(e))
      )
    }
  }
```

Reply:
Applied the suggestion in 7d08a89cf6d8dc047ccb0a01749e473046e60276, thanks.